syntax="proto3";
package communication_raft;
service MasterMapperCommunication
{
    rpc MapperParameters(MapRequest) returns (MapResponse);
}
service MapperReducerCommunication
{
    rpc PartitionParameters(PartionRequest) returns (PartionResponse);
}
service MasterReducerCommunication
{
    rpc ReducerParameters(ReducerRequest) returns (ReducerResponse);
}
message DataPoint
{
    float x=1;
    float y=2;
}
message length
{
    int32 startLength=1;
    int32 endLength=2;
}
message MapRequest 
{
    // int32 shardIndexStart =1;
    // int32 shardIndexEnd =2;
    repeated length Lengths = 1;
    repeated DataPoint CentroidCoordinates=2;
    int32 reducers=3;
}
message MapResponse
{
    bool success = 1;
}
message ReducerRequest
{
    repeated string mapperportnumbers=1;
}
message ReducerResponse
{
    string centroids=1;
    int32 success = 2;
    int32 mapper_id = 3;

}
message mappartion
{
    int32  key=1;
    DataPoint point=2;
    int32 freq=3;
}
message PartionResponse
{
    repeated mappartion mappartion=1;
}
message PartionRequest
{
    int32 reducerid=1;
}
